### PR TITLE
Revamps MetaStation Telecomms

### PR DESCRIPTION
**Pull request by AffectedArc07**
Old (Normal):
before

Old (Topology):
topology-old

New (Normal):
after

New (Topology):
topology-new

The piece of paper on the table in the middle is now a guide to telecomms. This is what the paper says:
-------------------------------------------------
**Congratulations On Your Purchase Of An NT-07 Telecommunications System.**
WARNING: Should this machinery become destroyed or non-functional, please contact Central Command Tech Support.
INFO: This machinery is very susceptible to ionspheric interference. Should an ionspheric anomaly occur, the machinery may be offline for a brief period of time.

List of authorised channels and corresponding frequencies:
Science: 135.1
Medical: 135.5
Supply: 134.7
Service: 134.9
Command: 135.3
Security: 135.9
Engineering: 135.7
Common: 145.7

The Topology:
This topology is split into aisles, each aisle having 1 bus and 1 CPU responsible for 2 channels, each with their own servers. The channel specific servers are marked with floor tiles similar to each channel identifier.

Telecommunication Machinery Functionality
All machines share similar features, such as networks and filtering frequencies, but each machine has a very specific role
Servers:
Servers store a data packet every time audio is sent down a corresponding channel, this data can be viewed with a Telecomms Server Monitor. Servers can also be used for scripting, but sadly this is no longer supported.

Hubs:
Hubs are the 'Hub' of the setup, quite literally. All the machine except processors (see below) link to this, including distant ones such as relays (See below).

Processors:
Processors are an essential part of a telecomms setup if you want to be able to understand anything. They take an encoded signal and decode it into standard audio which is sent off to the broadcasters (See below). These link to the bus mainframes.

Bus Mainframes
Bus Mainframes play a similar role to hubs. Networks without hubs will use a Bus Mainframe as a hub, as they can do local machinery but not relays. The bus links to all machines that cover the channels it manages. So to the hub, the processor and the server.

Receiver:
Receivers are what they say on the tin, they receive subspace signals. If you are using a hub, link it to the hub, if you are not, link it to the mainframe.

Broadcasters:
Broadcasters are what they say on the tin, they take decoded signals from the processors, and broadcast them through subspace. If you are using a hub, link it to the hub, if you are not, link it to the mainframe.

Other Machinery:
This machinery is not linked directly to the network, but is still important for communications.

PDA Server:
This is the server responsible for PDA messaging, if this is disabled, PDA messaging will not function.

Blackbox Recorder:
This is important should and disaster happen to the station, this keeps a record of everything that happens to the station.

NTNet Quantum Relay:
This is the machine which makes NTNet (Used by consoles and tablets) able to reach devices. Warning: This machine is very fragile.

This room is split into four isles, each one with a corresponding bus and CPU, as well as two servers.
Isle 1: Medical And Science
Isle 2: Service And Supply
Isle 3: Command And Security
Isle 4: Supply And Service
	
NOTE: In this setup, the Common server also listens on frequencies 144.4 - 148.7

Bottom Section:
In the bottom section, you will find two broadcasters (one for output, one for redundancy) and two receivers. The west receiver handles Science, Medical, Supply and Service. The east receiver handles Command, Security, Engineering, Common and all the other frequencies.
---------------------------------

I also made sprites for isles 1-8, to make the layout even better

:cl: AffectedArc07
add: A guide to the telecommunications setup now spawns in the middle of telecomms
tweak: Completely overhauled MetaStation telecomms, now it isnt complete spaghetti, only slight spaghetti
add: There are now sprites for isles 1-8
/:cl:

Please dont hurt me flattest, I will commit properly one day.

### CODE REVIEW

**Comment by davethwave** [comment]
Very impressive.

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -484,3 +484,66 @@
 /obj/item/paper/evilfax/proc/evilpaper_selfdestruct()
 	visible_message("<span class='danger'>[src] spontaneously catches fire, and burns up!</span>")
 	qdel(src)
+
+/obj/item/paper/telecomms
+	name = "Your Guide To Telecomms!"
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: you don't capitalise `to` in title-case

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -484,3 +484,66 @@
...
+	info = "<b>Congratulations On Your Purchase Of An NT-07 Telecommunications System!</b><br>\
+	WARNING: Should this machinery become destroyed or non functional, please contact your nearest technician.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `non functional` should be hyphened.

Comment (SamHPurp) [reply code/modules/paperwork/paper.dm]: `nearest` is odd language for an official document

File: code/modules/paperwork/paper.dm [hunk 3]
```diff
@@ -484,3 +484,66 @@
...
+	INFO: This machinery is very subsceptable to Ionspheric Interference. Should an Ionspheric Anomalie occur, the machinery may be offline for a short amount of time.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `subsceptable` -> `susceptible`

`Ionspheric` -> `ionospheric` and not capitalised `interference` either

`Ionspheric` -> `ionospheric` again. Also `Anomalie` -> `anomaly`

`short amount` -> `brief period`

File: code/modules/paperwork/paper.dm [hunk 4]
```diff
@@ -484,3 +484,66 @@
...
+	<li>Science: 135.1</li>\
+	<li>Medical: 135.5</li>\
+	<li>Supply: 134.7</li>\
+	<li>Service: 134.9</li>\
+	<li>Command: 135.3</li>\
+	<li>Security: 135.9</li>\
+	<li>Engineering: 135.7</li>\
+	<li>Common: 145.7</li><br>\
+	<br>\
+	This Topology:<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `The Topology`

File: code/modules/paperwork/paper.dm [hunk 5]
```diff
@@ -484,3 +484,66 @@
...
+	This topology is split into isles, each isle having 1 bus and 1 CPU responsible for 2 channels, each with their own servers. The channel specific servers are marked with floor tiles similar to each channel identifier.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `aisle` not `isle` x2

File: code/modules/paperwork/paper.dm [hunk 6]
```diff
@@ -484,3 +484,66 @@
...
+	<br>\
+	Telecommunication Machinery Description<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `Description` -> `Functionality`

File: code/modules/paperwork/paper.dm [hunk 7]
```diff
@@ -484,3 +484,66 @@
...
+	All machines share similar features, such as networks and filtering frequencies, but each machine has a very specific role<br>\
+	Servers:<br>\
+	Servers store a data packet every time audio is sent down a corresponding channel, this data can be viewed with a Telecomms Server Montior. Servers can also be used for scripting, but sadly this is no longer supported.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `Montior` -> `Monitor`

File: code/modules/paperwork/paper.dm [hunk 8]
```diff
@@ -484,3 +484,66 @@
...
+	<br>\
+	Hubs:<br>\
+	Hubs are the 'Hub' of the setup, quite literally. All the machine except processors (see below) link to this, including distant ones such as relays (See below).<br>\
+	<br>\
+	Processors:<br>\
+	Processors are an essential part of a telecomms setup if you want to be able to understand anything. They take an encoded signal, and decode it into standard audio which is sent off to the broadcasters (See below). These link to the bus mainframes.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: no comma after signal

File: code/modules/paperwork/paper.dm [hunk 9]
```diff
@@ -484,3 +484,66 @@
...
+	<br>\
+	Bus Mainframes<br>\
+	Bus Mainframes play a similar role to hubs. Networks without hubs will use a Bus Mainframe as a hub, as they can do local machinery but not relays. The bus links to all machines that cover the channels it manages. So to the hub, the processor and the server.<br>\
+	<br>\
+	Reciever:<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `Reciever` -> `Receiver`

File: code/modules/paperwork/paper.dm [hunk 10]
```diff
@@ -484,3 +484,66 @@
...
+	Recievers are what they say on the tin, they recieve subspace signals. If you are using a hub, link it to the hub, if you are not, link it to the mainframe.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `Reciever` -> `Receiver`

`recieve` -> `receive`

File: code/modules/paperwork/paper.dm [hunk 11]
```diff
@@ -484,3 +484,66 @@
...
+	<br>\
+	Broadcasters:<br>\
+	Broadcasters are what they say on the tin, they take decoded signals from the processors, and broadcast them through subspace. If you are using a hub, link it to the hub, if you are not, link it to the mainframe.<br>\
+	<br>\
+	Other Machinery:<br>\
+	This machinery is not linked directly to the network, but is still important for communications.<br>\
+	<br>\
+	PDA Server:<br>\
+	This is the server responsible for PDA messaging, if this is disabled, PDA messaging will fail to work, and thus not work.<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: lol

` if this is disabled, PDA messaging will fail to work, and thus not work.` -> ` if this is disabled, PDA messaging will not function.`

File: code/modules/paperwork/paper.dm [hunk 12]
```diff
@@ -484,3 +484,66 @@
...
+	<br>\
+	This room is split into four isles, each one with a corresponding bus and CPU, as well as two servers.<br>\
+	<li>Isle 1: Medical And Science</li>\
+	<li>Isle 2: Service And Supply</li>\
+	<li>Isle 3: Command And Security</li>\
+	<li>Isle 4: Supply And Service</li>\
+	<br>\
+	NOTE: In this setup, the Common server also listens on frequencies 144.4 - 148.7<br>\
+	<br>\
+	Bottom Section:<br>\
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: Weird language again

File: code/modules/paperwork/paper.dm [hunk 13]
```diff
@@ -484,3 +484,66 @@
...
+	In the bottom section, you will find two broadcasters (one for output, one for redundcancy) and two recievers. The west reciever handles Science, Medical, Supply and Service. The east reciever handles Command, Security, Engineering, Common and all the other frequencies.<br>"
```
Comment (SamHPurp) [inline code/modules/paperwork/paper.dm]: `redundcancy` -> `redundancy`

`recievers` -> `receivers`

`reciever` -> `receiver` x2

File: code/game/objects/structures/signs.dm [hunk 14]
```diff
@@ -328,4 +328,9 @@
 /obj/structure/sign/logo
 	name = "nanotrasen logo"
 	desc = "The Nanotrasen corporate logo."
-	icon_state = "nanotrasen_sign1"
\ No newline at end of file
+	icon_state = "nanotrasen_sign1"
+
+/obj/structure/sign/numbered/isle
```
Comment (SamHPurp) [inline code/game/objects/structures/signs.dm]: aisle

File: code/game/objects/structures/signs.dm [hunk 15]
```diff
@@ -328,4 +328,9 @@
 /obj/structure/sign/logo
 	name = "nanotrasen logo"
 	desc = "The Nanotrasen corporate logo."
-	icon_state = "nanotrasen_sign1"
\ No newline at end of file
+	icon_state = "nanotrasen_sign1"
+
+/obj/structure/sign/numbered/isle
+	name = "Isle Number"
```
Comment (SamHPurp) [inline code/game/objects/structures/signs.dm]: aisle

File: code/game/objects/structures/signs.dm [hunk 16]
```diff
@@ -328,4 +328,9 @@
 /obj/structure/sign/logo
 	name = "nanotrasen logo"
 	desc = "The Nanotrasen corporate logo."
-	icon_state = "nanotrasen_sign1"
\ No newline at end of file
+	icon_state = "nanotrasen_sign1"
+
+/obj/structure/sign/numbered/isle
+	name = "Isle Number"
+	desc = "This sign indicated an isle in a telecommunications setup."
```
Comment (SamHPurp) [inline code/game/objects/structures/signs.dm]: aisle

File: code/game/objects/structures/signs.dm [hunk 17]
```diff
@@ -328,4 +328,9 @@
...
 	name = "nanotrasen logo"
 	desc = "The Nanotrasen corporate logo."
-	icon_state = "nanotrasen_sign1"
\ No newline at end of file
+	icon_state = "nanotrasen_sign1"
+
+/obj/structure/sign/numbered/isle
+	name = "Isle Number"
+	desc = "This sign indicated an isle in a telecommunications setup."
+	icon_state = "islenum"
```
Comment (SamHPurp) [inline code/game/objects/structures/signs.dm]: aisle

**Comment by FlattestGuitar** [comment]
I have no idea how these work.

@SamHPurp, this one is yours.

File: code/game/objects/structures/signs.dm [hunk 18]
```diff
@@ -331,6 +331,6 @@
 	icon_state = "nanotrasen_sign1"
 
 /obj/structure/sign/numbered/isle
-	name = "Isle Number"
-	desc = "This sign indicated an isle in a telecommunications setup."
-	icon_state = "islenum"
\ No newline at end of file
+	name = "Aisle Number"
+	desc = "This sign indicated an aisle in a telecommunications setup."
+	icon_state = "islenum"
```
Comment (SamHPurp) [inline code/game/objects/structures/signs.dm]: icon state and the DMI too...

File: code/game/objects/structures/signs.dm (hunk 14 above)
Comment (SamHPurp) [reply code/game/objects/structures/signs.dm]: I swear to god AA

**Comment by AffectedArc07** [comment]
@SamHPurp

**Pull request merged**